src/components/Editor/tests/Editor.spec.js

```diff
@@ -42,38 +29,123 @@ function render(overrides = {}) {
+      expect(mockEditor.codeMirror.scrollTo.mock.calls).toEqual([]);
+    });
   });
 
-  it("should be loading", async () => {
-    const { component, mockEditor } = render();
-    await component.setState({ editor: mockEditor });
-    component.setProps({
-      selectedSource: I.fromJS({ loadedState: "loading" })
+  describe("When loaded", () => {
+    it("should show text", async () => {
+      const { component, mockEditor, props } = render({});
+
+      await component.setState({ editor: mockEditor });
+      await component.setProps({
+        ...props,
+        selectedSource: createMockSource({ loadedState: "loaded" }),
+        selectedLocation: { sourceId: "foo", line: 3, column: 1 }
+      });
+
+      expect(mockEditor.setText.mock.calls).toEqual([["the text"]]);
+      expect(mockEditor.codeMirror.scrollTo.mock.calls).toEqual([[1, 2]]);
     });
-    expect(mockEditor.setText.mock.calls).toEqual([["Loading…"]]);
   });
 
-  it("should set text", async () => {
-    const { component, mockEditor } = render();
-    await component.setState({ editor: mockEditor });
-    component.setProps({
-      selectedSource: I.fromJS({ text: "text change", loadedState: "loaded" })
+  describe("When navigating to a loading source", () => {
+    it("should show loading message and not scroll", async () => {
+      const { component, mockEditor, props } = render({});
+
+      await component.setState({ editor: mockEditor });
+      await component.setProps({
+        ...props,
+        selectedSource: createMockSource({ loadedState: "loaded" }),
+        selectedLocation: { sourceId: "foo", line: 3, column: 1 }
+      });
+
+      // navigate to a new source that is still loading
+      await component.setProps({
+        ...props,
+        selectedSource: createMockSource({
+          id: "bar",
+          loadedState: "loading"
+        }),
+        selectedLocation: { sourceId: "bar", line: 1, column: 1 }
+      });
+
+      expect(mockEditor.setText.mock.calls).toEqual([
+        ["the text"],
+        ["Loading…"]
+      ]);
+
+      expect(mockEditor.codeMirror.scrollTo.mock.calls).toEqual([[1, 2]]);
+    });
+  });
+
+  describe("When navigating to a loaded source", () => {
+    it("should show text and then scroll", async () => {
+      const { component, mockEditor, props } = render({});
+
+      await component.setState({ editor: mockEditor });
+      await component.setProps({
+        ...props,
+        selectedSource: createMockSource({ loadedState: "loading" }),
+        selectedLocation: { sourceId: "foo", line: 1, column: 1 }
+      });
+
+      // navigate to a new source that is still loading
+      await component.setProps({
+        ...props,
+        selectedSource: createMockSource({
+          loadedState: "loaded"
+        }),
+        selectedLocation: { sourceId: "foo", line: 1, column: 1 }
+      });
+
+      expect(mockEditor.setText.mock.calls).toEqual([
+        ["Loading…"],
+        ["the text"]
```

Comment:
same here -- it is unclear what relationship these text strings have to the expected behavior